### PR TITLE
infer from overloads: handle `/`

### PR DESCRIPTION
**Pull request by KotlinIsland**
- resolves: #367

### CODE REVIEW

File: test-data/unit/check-based-infer-function-types.test [hunk 1]
```diff
@@ -203,6 +203,32 @@ def f(**kwargs):  # E: Function is missing a type annotation for one or more arg
...
+def bar(z, b):
+    reveal_type(z)
+    reveal_type(b)
+[out]
+main:4: error: invalid syntax  [syntax]
+[out version>=3.8]
+main:9: note: Revealed type is "int | str"
+main:17: note: Revealed type is "int | None"
+main:18: note: Revealed type is "str | ellipsis"
+
```
Comment (DetachHead) [inline test-data/unit/check-based-infer-function-types.test:231]: add tests for `*` as well? and a test for if one overload has a `/` and the other doesnt? etc

Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:267]: done?

File: test-data/unit/check-based-infer-function-types.test [hunk 2]
```diff
@@ -203,6 +239,32 @@ def f(**kwargs):  # E: Function is missing a type annotation for one or more arg
...
+@overload
+def bar(x: int, /, b: str): ...
+@overload
+def bar(y: None, /, b: ellipsis): ...
+
+def bar(z, b):
+    reveal_type(z)
+    reveal_type(b)
+[out]
+main:4: error: invalid syntax  [syntax]
```
Comment (DetachHead) [inline test-data/unit/check-based-infer-function-types.test:262]: invalid syntax?

Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:262]: `/` is invalid syntax < 3.8

File: test-data/unit/check-based-infer-function-types.test (hunk 1 above)
Comment (DetachHead) [reply test-data/unit/check-based-infer-function-types.test:267]: hmm, what happens if the `/` or `*` is in a different position in each overload

File: test-data/unit/check-based-infer-function-types.test (hunk 2 above)
Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:262]: @DetachHead I think this branch is complete and can be merged, just this thread remaining

File: test-data/unit/check-based-infer-function-types.test (hunk 1 above)
Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:285]: added

Comment (DetachHead) [reply test-data/unit/check-based-infer-function-types.test:285]: did u push? i don't see it

Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:285]: https://github.com/KotlinIsland/basedmypy/compare/753aa264c8dd4316c5d7d20f9184e95f5bd57a2d..95acb2286b1b4506dfdabbb2b49e38b07cc0b4b8

Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:285]: or did you mean something else?

Comment (DetachHead) [reply test-data/unit/check-based-infer-function-types.test:285]: thats a good one too, tho i meant something like
```py
@overload
def f(x: int, /, b: int): ...
@overload
def f(x: int, b: int, /): ...
```

Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:285]: sure

Comment (KotlinIsland) [reply test-data/unit/check-based-infer-function-types.test:285]: nah

**Review by DetachHead — APPROVED** [review]

**Pull request merged**